### PR TITLE
Added Donksoft Toy Vendor (Machine Board) to the Circuit Imprinter [Fixed PR]

### PR DESCRIPTION
**Pull request by Kerbin-Fiber**
:cl: Kerbin_Fiber
add: Donksoft Toy Vendor Boards to the Circuit Imprinter when Illegal tech is high enough.
add: Donksoft Toy Vendor Restocking Units come in Toy Crates from Cargo!
/:cl:
donksoftrestockingunit

Why? Because it's no fair for the Syndicates to be able to have ~~fun~~ toys and we don't! Once the research from illegal items that were ~~stolen~~ obtained have been deconstructed, the Circuit Imprinter will be able to print out Donksoft Toy Vendor (Machine Board)s!

### CODE REVIEW

**Comment by WJohn** [comment]
absolute madman 🥇

**Review by FantasticFwoosh — COMMENTED** [review]
Ok the donksoft changes are apparent, but please re-fork and try again because opening statement and detail of what you intended to do is 👌  but the commit details are still borked  :dog:

- A easy way to clean a remote is to delete your repo from web-side git via settings, scrolling to the bottom and deleting, then running all they way back to TG github and hitting that fork button, there are fancier high tech ways of doing it but that's a bog standard way of doing it.

So your version of TG station is up to date BEFORE commiting your changes, as you can see there's a lot of extra stuff not relevant to the PR pulled along because your repository stored in your profile is older than TG's current state.

**Comment by FantasticFwoosh** [comment]
Can you not remove that changelog to make a cleaner PR?

**Comment by GunHog** [comment]
I never looked into how Cargo randomized crates work. I need three stocking units to build the Donksoft vendor, yes?

I glanced at the code, and I understand it to be that it picks five different items from the list to fill the crate. So, I basically have a one-in-three (5 items from a list of 15) chance of getting a donksoft stocking unit. I need a minimum of three crates to get a restocking unit. Given these odds, I imagine that if I actually want one of these things, I am probably going to end up filling the cargo bay with toy crates.

Still, if someone manages to build one or two, I would love to read the story about it.

**Comment by Armhulen** [comment]
The donk soft vendor requires a syndicate id, you would have to remove that.

File: html/changelogs/AutoChangeLog-pr-30416.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+author: "Xhuis"
```
Comment (Cyberboss) [inline html/changelogs/AutoChangeLog-pr-30416.yml]: Remove this file from your PR

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by ma44** [comment]
## As the original creator of this, please for the love of god don't add this in unless you want to drastically buff science and cargo while also making antags and station crew to engage in a powergaming war of who gets this stuff first

'But ma44 don't you like fun stuff?'
Yes fun stuff like this is great, but it shouldn't be **THIS** easily gotten, let's look at the things wrong with this!

- Riot darts, while having a robust supply of **400** reusable riot darts (that can also be used for metal in a autolathe and recycled as well) as well as all of the guns that have riot darts preloaded. This was meant to make the syndicate base never run out of riot darts for whatever reason, and that they also don't have a autolathe

- Alright let's look at the guns because they are obviously the main reason on why this thing is bad for the station

1. The pistol isn't that horribly bad, pretty sure you can get it from somewhere
2. Shotgun is also not that great, only carries 4 shots and requires a pump per shot, usually not enough to stun a person
3. Crossbow, same as shotgun but holds 5 darts
4. The C20r, holds 20 darts, easy to use magazines, fires two darts at a time and fits in a backpack. This is well literally a burst disabler with nearly **infinite** ammo. Remember to duel wield it so you can fire four darts in a goddamm volley. 
5. The l6 saw, 50 rounds and a triple burst. While this is painfully obvious to anyone that have this out, it still has a crapton of suppression it can output. 
6. The SMG, meant to represent the old SABR SMG, comes with **42** rounds and a goddamm **triple burst**. Literally duel wielding this can make you stamina stun someone in a single goddamm volley and suppresses harder than a goddamm l6 saw. As a bonus this can fit in backpacks.

- The origin tech of the circuit board can make RnD unlock large energy crossbows as well as suppressors while only needing to raise the syndicate tech to 2 (usually achievable with low cost TC items, RnD magicry or a TYRANT AI module circuit) Man, so many stuns!

- Op without adrenalins are going to die so hard, even if they do they can be kept down with all of the ammo powergamers can carry. 

# TLDR merge this if you want burst disablers from gun crafting back, but now the entire station can be equipped with this

**Comment by GunHog** [comment]
Geez, you could at least let it ruin one round before you drop an accurate and convincing, if somewhat buzzwordy, argument against it. I would have liked to participate in a station-wide deathmatch. At least one... :sad:

**Comment by ma44** [comment]
1. apply for admin
2. spawn this in
3. ?
4. Enjoy deathmatch

**Comment by The-clown-known-as-Pepe** [comment]
>this cancer almost every round

no

**Comment by Kerbin-Fiber** [comment]
Many people don't want this to happen and others do, I suggest we actually try it out before we throw it down the drain, yes it sounds very biased but still we can at least try it. I've taken into account of how many times I've seen Science have Illegal Tech over 1 and it's very rare for someone to actually do it from what I know of. Gunhog did see that you also have a low chance of getting the restocking units from cargo, it kinda cancels out the point of "Taking down operartives with riot foam darts" when you need to be lucky enough to even get the resupply units before the Ops arrive. With the chances so low to even make it (And needing it to be hacked so everyone can use it (Purple light > Yellow)), it seems like it would only be used on extended rounds. Rushing to get the tech for the board is eh but getting the crates from cargo and wasting lot of credits is another factor to take in. Please at least try this before thrown down the drain.

File: html/changelogs/AutoChangeLog-pr-30416.yml (hunk 1 above)
Comment (Kerbin-Fiber) [reply html/changelogs/AutoChangeLog-pr-30416.yml]: It'd help if you told me how to, since I am new to using this Git Tortoise thing.

**Comment by ma44** [comment]
Any cargonia that tries hard enough can easily get hundreds of thousands of credits just in 20-30 minutes (or maybe just below 100k, as that should be enough for your cool little vendor) 

This would also encourage scientists to go git that illegal tech up so they can enjoy crossbows or some cool pseudo burst disablers.

**Comment by BlueNothing** [comment]
Would it be good to encourage scientists to do that?

**Comment by Kerbin-Fiber** [comment]
I upped the tech level for it so it should encourage them to get even more illegal tech if they want it that bad.

**Comment by PKPenguin321** [comment]
I would put the board on a space ruin instead of in R&D, would practically guarantee it couldnt be powergamed while still potentially letting it happen now and then

**Comment by BlueNothing** [comment]
Between the high tech level requirements and the non-trivial cost to
cargo... it might or might not. RnD or Mining would definitely have to kick
in hard, and this would almost certainly be instead of null crates to some
significant extent... but how often do people ruin-dive?

**Comment by Kerbin-Fiber** [comment]
so here are the reasons why this is a viable idea:
* You have to get co-operation of TWO departments (Not any greyshitter can get that to happen)
* Science actually has to work to get the tech high enough
* Even if cargo complies, there will be some problems getting all of the restocking units
* When complete, people can make rooms into Foam Force arenas, ~~**Enables FUN**~~

**Comment by BlueNothing** [comment]
Re 1: If science is half-decent, cargo is sane, mining comes in with trace
amounts of everything or more, RnD is as good as maxed and cargo can be
assumed to have an arbitrary number of points. To wit, who said greys are
going to be the ones trying to get it? I'd fully expect cargo or sci to be
the ones to try it.

Re 2: There's a good chance that RnD is being done by someone competent,
probably 50/50 odds or better, meaning that unless mining is dead or
decided to screw off and hunt megafauna from the word go, RnD will max out
within 30 minutes at least half the time. From there, cargo has 1-2 hundred
thousand cargo points to play with. Sec loves disablers, cargo would likely
use them in place of shotguns unless megafauna, etc, are expected. So
Cargo's likely rotation: Insulated gloves, toolbelt crate, maybe get some
medical or pin crates, then wait for a big tech disk. From there... expect
either loads of nullcrates or loads of toy crates. That said... the
increased Illegal tech requirement means these will be made only quite
rarely, because if you can *get* them, you can very probably get nullcrates.

**Comment by BlueNothing** [comment]
That's why I'd suggest Illegal 2. Then again, there's a significant chance
that it'll either pretty much always happen (tech is low, can be achieved
by a sleepwalking RnD specialist), or never happen (if you can get it, you
won't, because cargo is probably emagged if you have Illegal 4 anyway,
meaning that the shuttle will be full of straight nullcrates, since any
traitor with half a brain will get an emag if they can afford it unless
they're going full murderbone).

**Comment by ma44** [comment]
This is going to be one of those features that gets a revert pr in a month because some powergamer decides to build this nearly every round for cool chaos literally everywhere

**Comment by GunHog** [comment]
> This is going to be one of those features that gets a revert pr in a month because some powergamer decides to build this nearly every round for cool chaos literally everywhere

Which of course would be you, because powergamers sometimes abuse things as a statement to get them removed!

**Comment by BlueNothing** [comment]
Case in point: Vcordie and Telesci.

**Comment by GunHog** [comment]
After being educated by @ma44 about riot darts, why not just remove those from the vendor? 

You could have normal darts only affect people with a laser tag vest (or a new Donk vest / toy armor?) so people with 'burst disablers' can only stun other Donksoft/laser tag players. 

Note: Ma44 did not explain how else people acquire riot darts, so I do not know how viable an idea this is.

**Comment by BlueNothing** [comment]
Hacked autolathes also make them, but it could be okay if they're Illegal 2
but don't come with ammo?

**Comment by GunHog** [comment]
Hacked autolathes? You would have to remove that, then.

**Comment by BlueNothing** [comment]
If you make it contingent on wearing a vest, mostly nobody's going to wear such a vest.

**Comment by Kerbin-Fiber** [comment]
Alright, I'll change it so that it doesn't give out riot foam darts, or at least to the best of my abilities.

**Comment by ma44** [comment]
The fluke awps donksoft guns start off with riot darts. Also note that some of the guns can also prespawn with riot darts.

**Comment by Ratvar** [comment]
animated/vendorrisen donksoft should shoot people with riot darts down and glomp up in close range
totally

**Comment by juanototo2** [comment]
Dude the Syndicate lavabase needs riot darts if you can just make it a subtype without the riots instead jeez

**Comment by Kerbin-Fiber** [comment]
That's what I'm doing. I'm removing all riot foam darts from the vendor by making it a subtype.

**Comment by ma44** [comment]
May as well rename this pr 'removes riot darts from the autolathe while adding in a cool toy gun vendor that doesn't have riot darts'

**Comment by Kerbin-Fiber** [comment]
I said I'm not touching the Autolathe, this PR is for the Vendor only.
